dto/case_event_dto.go

```diff
@@ -4,16 +4,17 @@ import (
 	"time"
 
 	"github.com/checkmarble/marble-backend/models"
+	"github.com/guregu/null/v5"
 )
 
 type APICaseEvent struct {
-	Id             string    `json:"id"`
-	CaseId         string    `json:"case_id"`
-	UserId         string    `json:"user_id"`
-	CreatedAt      time.Time `json:"created_at"`
-	EventType      string    `json:"event_type"`
-	AdditionalNote string    `json:"additional_note"`
-	NewValue       string    `json:"new_value"`
+	Id             string      `json:"id"`
+	CaseId         string      `json:"case_id"`
+	UserId         null.String `json:"user_id"`
```

Comment:
IIUC, this is the change that make `user_id` optionnal in the returned JSON payload, is that right ?

Reply:
(I ask because of [this line in the pkg readme](https://github.com/guregu/null?tab=readme-ov-file#bugs))

Reply:
Yes, that makes it so that a null input (aka value.Valid = false) will encode to JSON `null` while an empty string (`value.valid=true, value.String=""`) will encode to JSON `""`.
I could also get it to completely omit the user_id field in the output if empty string is received by marking `json:"user_id,omitempty"` but I like to have nulls present if they're in the schema